### PR TITLE
addind CUDA BW test and custom test set up script

### PR DESCRIPTION
**Pull request by rafsalas19**
- Create setup script which compiles the CUDA BW test executable and copies all nhc test in the customTests directory to /etc/nhc/scripts directory
- Adding custom test for Cuda bandwidth. to use just add ``` * || check_cuda_bw``` to conf file
   - gpu-copy.cu source file
   - azure_cuda_bandwidth.nhc

### CODE REVIEW

File: customTests/azure_cuda_bandwidth.nhc [hunk 1]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (rajvi) [inline customTests/azure_cuda_bandwidth.nhc:1]: Are these tested to see if they run as expected?

File: customTests/azure_cuda_bandwidth.nhc [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    fi
+
+	#Count the GPUs.
+	gpu_list="timeout 3m nvidia-smi --query-gpu=name --format=csv,noheader"
+    if ! catch_error "$gpu_list" "$error_smi"; then
+        return 1
+    fi
+	ngpus=$(echo "$output" | wc -l)
+
+	#Run the superbench device to host bandwidth test.
```
Comment (moes1) [inline customTests/azure_cuda_bandwidth.nhc]: Maybe say that it is running the device to host bandwidth test based on the one in superbench. I'm not sure it's exactly the same any more.

File: customTests/azure_cuda_bandwidth.nhc [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	exec_htod="timeout 3m $EXE_DIR/gpu-copy --size 134217728 --htod"
+	error_htod="**Fail** The htod gpu_copy test failed to execute."
+	error_htod+="It exited with error code"
+	if ! catch_error "$exec_htod" "$error_htod"; then
+        return 1
+    fi
+	x_htod=$(echo "$output")
+	
+
+	#Run the superbench host to device bandwidth test.
```
Comment (moes1) [inline customTests/azure_cuda_bandwidth.nhc]: same comment as above here.

File: customTests/azure_cuda_bandwidth.nhc (hunk 1 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:1]: This was tested on ND96asr and has the expected 24 GBS BW. i.e
  * || check_cuda_bw 24 passes 
  
  but let me verify the test fails on the right condition

File: customTests/azure_cuda_bandwidth.nhc (hunk 2 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: Let me address

File: customTests/azure_cuda_bandwidth.nhc (hunk 3 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: let me address

File: customTests/azure_cuda_bandwidth.nhc [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		#If the min bandwidth is too low the test has failed.
+		if [ $min_bw -lt $EXP_CUDA_BW ]; then
+			die "Bandwidth is low on device $i. Reported bandwidth is"\
+				"$min_bw GB/s."
+			pass=0
+			return 1
+		fi
+	done
+	#Unlock the graphics clock.
+	unlock_clocks="sudo timeout 3m nvidia-smi -rgc"
```
Comment (moes1) [inline customTests/azure_cuda_bandwidth.nhc:122]: Should we write a special catch_error code for use in the section while gpu clocks have been locked that will unlock the gpu clock before exiting the code?

Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:122]: Thats a really good idea. Let me implement!

Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:122]: Addressed

File: customTests/azure_cuda_bandwidth.nhc (hunk 1 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc:1]: Test is working as expected

File: customTests/azure_cuda_bandwidth.nhc (hunk 2 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: removed mention of superbench

File: customTests/azure_cuda_bandwidth.nhc (hunk 3 above)
Comment (rafsalas19) [reply customTests/azure_cuda_bandwidth.nhc]: removed mention of superbench

**Review by moes1 — APPROVED** [review]

**Pull request merged**